whole-tale/dashboard
# Refactor Run View to use taleId instead of instanceId

### Problem
We don't currently have an integrated view of a Tale's files and metadata without running an Instance of the Tale. This is unnecessary, since the filesystem and metadata exist independently of the running container, and could be operated on separately.

### Approach
The `/run/:id` view currently takes an optional instance ID, which assumes that we have a running container. If this view instead were to take a Tale ID, we can perform a lookup to see if running Instance exists for the Tale.

This presents a few new cases for the Run view, detailed below.

#### Running Tale (Instance Exists, and is `Running`)
* The current run view, more or less
* Interact tab should display an iframe or button to open in new tab
* Additional "Stop" button added at the top-right

#### Stopped Tale (No Instance Exists)
* The current run view, more or less
* Interact tab should display a placeholder message
* Additional "Start" button added at the top-right

#### Transitioning Tale (Instance Exists, but status is not `Running`)
* The current run view, more or less
* Interact tab should display a placeholder message and loading indicator
* Disabled "Please wait" button added at the top-right - this button re-enables itself ~30s after submitting a start or stop request to allow the user to attempt to submit a stop operation.

### How to Test
1. Checkout and run this branch locally, rebuild the dashboard
2. Close all incognito windows and, using a fresh one, login to the WholeTale Dashboard
    * You should immediately notice that the right-hand panel for "Launched Tales" is now gone
3. Compose a Tale if you haven't already
4. From the `Browse` page, click "View" for a Tale
    * You should no longer instead be brought to "Tale View" page when clicking this link
    * You should instead be brought to `Run` view for the Tale
5. Make sure the Tale is stopped - click `Stop` at the top-right, if necessary
    * With a stopped Tale, you should see a placeholder message on Interact, but still be able to use the Files/Metadata tabs
6. Click `Start` at the top-right to Start this Tale
    * You should see the "Start" button change to "Please Wait" with a loading indicator
    * You should see the placeholder message change to "Please Wait" with a loading indicator
    * You should see event notifications coming through, indicating that your Tale image is being rebuilt
7. Select the `Interact` tab and wait for the Tale to finish starting
    * Once the Tale is started, you should see the "Please Wait" on the button at the top-right has changed to "Stop"
    * If the chosen Tale uses an iframe, you should see the placeholder message has been replaced with an iframe to the Tale
    * If the chosen Tale doesn't use an iframe, you should see a placeholder message indicating that the user needs to access the Tale using a new browser tab
    * You should still be able to use the Files / Metadata tabs as before
8. Click `Stop` at the top-right
    * You should see the original placeholder message return
    * You should still be able to use the Files / Metadata tabs as before

#### TODOs
* The "Compose" flow no longer needs to start the Tale for the user upon creation - it is unclear whether the UI or the API currently handles this
* We should now be free to clean-up massive amounts of code from the UI (e.g. defunct controllers, old components, unreachable routes/views, etc)
